src/v2i-hub/MapPlugin/src/MapPlugin.cpp

```diff
@@ -342,43 +245,64 @@ bool MapPlugin::LoadMapFiles()
 					else if (inType == "TXT")
 					{
 						byte_stream bytes;
-						ifstream in(fn);
-						in >> bytes;
-
-						PLOG(logINFO) << fn << " MAP encoded bytes are " << bytes;
+						std::ifstream in(fn, std::ios::binary);
+						if (!in) {
+							PLOG(logERROR) << "Failed to open file: " << fn;
+						} else {
+							try {
+								std::string fileContent((std::istreambuf_iterator<char>(in)), std::istreambuf_iterator<char>());
+								in.close();
+								// Remove any newline characters
+								fileContent.erase(remove(fileContent.begin(), fileContent.end(), '\n'), fileContent.end());
+
+								// Check for and remove MessageFrame
+								if (fileContent.size() >= 4 && fileContent.substr(0, 4) == "0012") {
```

Comment:
What happens if there are multiple "38" in the binary payload? The same for 0012?

Reply:
Great findings on these bugs!

Reply:
@dan-du-car The message I used had 6 "38" in it. It only looked for the first instance, so this worked. But I can change it to make sure it only looks for it in the first couple of bytes depending on payload length.

Reply:
I'll leave it as is. Find only looks for the first instance of the string.